flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -48,144 +39,157 @@
-          Schema.Parser parser = new Schema.Parser();
-          InputStream is = null;
-          try {
-            FileSystem fs = FileSystem.get(URI.create(url), conf);
-            if (url.toLowerCase(Locale.ENGLISH).startsWith("hdfs:/")) {
-              is = fs.open(new Path(url));
-            } else {
-              is = new URL(url).openStream();
-            }
-            return parser.parse(is);
-          } finally {
-            if (is != null) {
-              is.close();
-            }
-          }
-        }
-      });
+  private boolean reuseEntity = true;
 
+  /**
+   * The Flume transaction. Used to keep transactions open across calls to
+   * process.
+   */
+  private Transaction transaction = null;
+
+  // Factories
+  private EntityParserFactory parserFactory = new EntityParserFactory();
+  private FailurePolicyFactory policyFactory = new FailurePolicyFactory();
+
+  /**
+   * Return the list of allowed formats.
+   * @return The list of allowed formats.
+   */
   protected List<String> allowedFormats() {
     return Lists.newArrayList("avro", "parquet");
   }
 
   @Override
   public void configure(Context context) {
+    this.context = context;
+
     // initialize login credentials
     this.login = KerberosUtil.login(
-        context.getString(DatasetSinkConstants.AUTH_PRINCIPAL),
-        context.getString(DatasetSinkConstants.AUTH_KEYTAB));
-    String effectiveUser =
-        context.getString(DatasetSinkConstants.AUTH_PROXY_USER);
+        context.getString(AUTH_PRINCIPAL),
+        context.getString(AUTH_KEYTAB));
+    String effectiveUser
+        = context.getString(AUTH_PROXY_USER);
     if (effectiveUser != null) {
       this.login = KerberosUtil.proxyAs(effectiveUser, login);
     }
 
-    String datasetURI = context.getString(
-      DatasetSinkConstants.CONFIG_KITE_DATASET_URI);
+    // Get the dataset URI and name from the context
+    String datasetURI = context.getString(CONFIG_KITE_DATASET_URI);
     if (datasetURI != null) {
-      this.target = URI.create(datasetURI);
-      this.datasetName = uriToName(target);
+      this.datasetUri = URI.create(datasetURI);
+      this.datasetName = uriToName(datasetUri);
     } else {
-      String repositoryURI = context.getString(
-        DatasetSinkConstants.CONFIG_KITE_REPO_URI);
-      Preconditions.checkNotNull(repositoryURI, "Repository URI is missing");
-      this.datasetName = context.getString(
-        DatasetSinkConstants.CONFIG_KITE_DATASET_NAME);
-      Preconditions.checkNotNull(datasetName, "Dataset name is missing");
-
-      this.target = new URIBuilder(repositoryURI, datasetName).build();
+      String repositoryURI = context.getString(CONFIG_KITE_REPO_URI);
+      Preconditions.checkNotNull(repositoryURI, "Repository URI is missing."
+          + " You must set either " + CONFIG_KITE_DATASET_URI + " or both"
+          + " " + CONFIG_KITE_REPO_URI + " and " + CONFIG_KITE_DATASET_NAME);
```

Comment:
I think this should only encourage the user to set the dataset URI because the other settings are deprecated.